enumeratum-core/src/main/scala/enumeratum/Enum.scala

```diff
@@ -38,20 +39,20 @@ trait Enum[A <: EnumEntry] {
   /**
     * Map of [[A]] object names to [[A]]s
     */
-  lazy final val namesToValuesMap: Map[String, A] =
-    values.map(v => v.entryName -> v).toMap
+  lazy val namesToValuesMap: Map[String, A] =
+    values.map(v => v.entryName -> v)(breakOut)
```

Comment:
I'm not a fan of using `breakOut` here; I know what it does, but:

  1. It's not obvious to beginners how it works; 
  2. It'll likely be gone with the [new collections lib](https://www.scala-lang.org/blog/2017/02/28/collections-rework.html)
  3. This is a cached value; the so gains are pretty minimal.

Reply:
tl;dr Removed `breakOut`.

`breakOut` is definitely complex since `CanBuildFrom` isn't exactly the most intuitive concept, and I know that its days are numbered in Scala collections--though it isn't deprecated and probably won't be for some time. It just won't be necessary with the new views. 

Also, `breakOut` is a memory optimization to eliminate the need for an intermediate sequence--cached values or not. I worry that might matter when people use enums in "Big Data" (as with Apache Spark) scenarios, but it is definitely true that the sequences for most cases are so small that `breakOut` probably doesn't change the world.